samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemProducer.scala

```diff
@@ -98,14 +80,13 @@ class KafkaSystemProducer(systemName: String,
 
   def register(source: String) {
     if(sources.putIfAbsent(source, new SourceData) != null) {
-      throw new SamzaException("%s is already registered with the %s system producer" format (source, systemName))
+      throw new SystemProducerException("%s is already registered with the %s system producer" format (source, systemName))
     }
   }
 
   def closeAndNullifyCurrentProducer(currentProducer: Producer[Array[Byte], Array[Byte]]) {
```

Comment:
Minor, Unrelated to PR: Personally prefer placing private/helper methods after public methods (or methods that use them). +1 for updating if you agree ;)